engine/src/org/pentaho/di/trans/steps/pentahoreporting/PentahoReportingOutput.java

```diff
@@ -221,33 +225,54 @@ private void processReport( Object[] r, String sourceFilename, String targetFile
         }
       }
 
-      switch ( outputProcessorType ) {
+      ModifiableConfiguration modifiableConfiguration = (ModifiableConfiguration) report.getConfiguration();
+      String property;
+      Runnable exportTask;
+      switch( outputProcessorType ) {
```

Comment:
Since the "property" variable isn't used outside of this switch/case, what do you think about moving all these strings to constants in the corresponding ExportTask classes?

Also, since the modifiableConfiguration is retrieved from the report object, and the report object is consistently passed in to the ExportTasks, could we maybe just set that property in the constructor of the ExportTask which would reduce this switch/case to simply selecting the correct task?